### PR TITLE
coal mining example added to TensorMechanics.

### LINKED ISSUE #9308: Add mining example to TensorMechanics
### Description of the enhancement or error report

Add an example to TensorMechanics that shows how it can be used for simulation of mining.  Just a simple 2D example.

### Rationale for the enhancement or information for reproducing the error

This will help my colleagues who are following in my MOOSE footsteps and it will demonstrate some aspects of TensorMechanics that are useful for modelling geomaterials.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

Example only

### PR DESCRIPTION
**Pull request by WilkAndy**
No code here, just doco and a couple of heavy tests.

Fixes #9308

### CODE REVIEW

**Comment by dschwen** [comment]
Can you make sure you add all the pdf stuff using git-lfs? Also having the latex source _and_ the final document in the repo seems a bit redundant.

File: modules/tensor_mechanics/examples/coal_mining/tests [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    input = 'cosserat_elastic.i'
+    csvdiff = 'cosserat_elastic.csv'
+    heavy = true
+  [../]
+  [./cosserat_wp_only]
+    type = RunApp
+    input = 'cosserat_wp_only.i'
+    csvdiff = 'cosserat_elastic.csv'
+    heavy = true
+  [../]
```
Comment (dschwen) [inline modules/tensor_mechanics/examples/coal_mining/tests]: I suggest using a syntax only test here like we do for all other examples.

File: modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i [hunk 2]
```diff
@@ -0,0 +1,569 @@
+# Strata deformation and fracturing around a coal mine
+#
+# A 2D geometry is used that simulates a transverse section of
+# the coal mine.  The model is actually 3D, but the "x"
+# dimension is only 10m long, meshed with 1 element, and
```
Comment (dschwen) [inline modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i]: out of curiosity, why is this not a 2D mesh then?

**Comment by WilkAndy** [comment]
Could you instruct me how to do the git-lfs ?

I want to keep both the .tex and the .pdf .  The former is needed when people want to modify, and the latter is very convenient.

File: modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i (hunk 2 above)
Comment (dschwen) [inline modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i]: is that a cosserat thing (pardon my ignorance)?

Comment (WilkAndy) [inline modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i]: Most of the plasticity models that i intend on exploring in this example are only formulated in 3D (DruckerPrager and MohrCoulomb).  I wanted to keep the same setup throughout all the simulations.

Comment (WilkAndy) [inline modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i]: Oh yea, the LayeredCosserat that i use is also only for 3D - i forgot about that.

Comment (dschwen) [inline modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i]: But, the stress and strain tensors are 3D in either case, aren't they?

**Comment by dschwen** [comment]
https://www.atlassian.com/git/tutorials/git-lfs#tracking-files

You can probably do `git lfs track *.pdf` in the example directory.

File: modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i (hunk 2 above)
Comment (WilkAndy) [inline modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i]: I also need to set the stress in the "third" direction and keep the strain in that direction fixed to zero.  Well, i guess i don't **need** to, but it's instructive if i do.

**Comment by WilkAndy** [comment]
Hmmm, just looked at this - i don't have git-lfs on the computer i'm using right now.  annoying!

File: modules/tensor_mechanics/examples/coal_mining/tests (hunk 1 above)
Comment (WilkAndy) [inline modules/tensor_mechanics/examples/coal_mining/tests]: OK, i can do the syntax test only.  But i do have some CSV data that i can compare with, so do you think two tests: one syntax and the other "heavy" is a good idea?  The "heavy" test isn't incredibly lengthy, but it does take over 2 seconds.

File: modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i (hunk 2 above)
Comment (WilkAndy) [inline modules/tensor_mechanics/examples/coal_mining/cosserat_elastic.i]: I just spent a while trying to formulate this in 2D.  The Cosserat stuff just isn't allowing me to do it.  I just set disp_x as an AuxVariable (and always zero) so i haven't got those disp_x DOFs hanging around at least.

**Comment by dschwen** [comment]
Andy, you'll have to squash the commits, otherwise the heavy files are still in teh git history, defeating the point of using LFS.

**Comment by dschwen** [comment]
Your patch does not add the files to LFS. I could try my luck and fix this if you'd like.

**Comment by dschwen** [comment]
This commit: 5802cbdea0 (in https://github.com/dschwen/moose/tree/coal_mining_eg_9308) does the job. I suggest closing this PR and letting me open a replacement PR.

**Comment by WilkAndy** [comment]
OK, thanks @dschwen .  Very strange - i was pretty sure i got a positive response back from git-lfs.

**Comment by dschwen** [comment]
Replaced by #9331

**Pull request closed without merging**